homeassistant/components/binary_sensor/bbb_gpio.py

```diff
@@ -0,0 +1,103 @@
+"""
+Support for binary sensor using Beaglebone Black GPIO.
+
+Example configuration:
+
+binary_sensor:
+  - platform: bbb_gpio
+    pins:
+      P8_12:
+        name: Door
+      GPIO0_26:
+        name: Window
+        bouncetime: 100
+        invert_logic: true
+        pull_mode: DOWN
+"""
+import logging
+
+import voluptuous as vol
+
+import homeassistant.components.bbb_gpio as bbb_gpio
+from homeassistant.components.binary_sensor import (
+    BinarySensorDevice, PLATFORM_SCHEMA)
+from homeassistant.const import (DEVICE_DEFAULT_NAME, CONF_NAME)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DEPENDENCIES = ['bbb_gpio']
+
+CONF_PINS = 'pins'
+CONF_BOUNCETIME = 'bouncetime'
+CONF_INVERT_LOGIC = 'invert_logic'
+CONF_PULL_MODE = 'pull_mode'
+
+DEFAULT_BOUNCETIME = 50
+DEFAULT_INVERT_LOGIC = False
+DEFAULT_PULL_MODE = 'UP'
+
+PIN_SCHEMA = vol.Schema({
+    vol.Required(CONF_NAME): cv.string,
+    vol.Optional(CONF_BOUNCETIME, default=DEFAULT_BOUNCETIME): cv.positive_int,
+    vol.Optional(CONF_INVERT_LOGIC, default=DEFAULT_INVERT_LOGIC): cv.boolean,
+    vol.Optional(CONF_PULL_MODE, default=DEFAULT_PULL_MODE):
+        vol.In(['UP', 'DOWN'])
+})
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_PINS, default={}):
+        vol.Schema({cv.string: PIN_SCHEMA}),
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Beaglebone Black GPIO devices."""
+    pins = config.get(CONF_PINS)
+
+    binary_sensors = []
+
+    for pin, params in pins.items():
+        binary_sensors.append(BBBGPIOBinarySensor(pin, params))
+    add_devices(binary_sensors)
+
+
+class BBBGPIOBinarySensor(BinarySensorDevice):
+    """Represent a binary sensor that uses Beaglebone Black GPIO."""
+
+    def __init__(self, pin, params):
+        """Initialize the Beaglebone Black binary sensor."""
+        # pylint: disable=import-error,no-member
+        import Adafruit_BBIO.GPIO as GPIO
+        self._pin = pin
+        self._name = params.get(CONF_NAME) or DEVICE_DEFAULT_NAME
+        self._bouncetime = params.get(CONF_BOUNCETIME)
+        self._pull_mode = params.get(CONF_PULL_MODE)
+        self._invert_logic = params.get(CONF_INVERT_LOGIC)
+
+        bbb_gpio.setup_input(self._pin, self._pull_mode)
+        self._input = bbb_gpio.read_input(self._pin)
+        self._state = True if self._input is GPIO.HIGH else False
```

Comment:
Let's modify [`read_input`](https://github.com/home-assistant/home-assistant/blob/2aa996b55820ef71367e0957701272c4ef44ab1d/homeassistant/components/bbb_gpio.py#L66) to `GPIO.input(pin) is GPIO.HIGH`, that way we're getting a boolean back. We'll match the interface or RPIO that way, we won't have to import Adafruit_BBIO here, and we can reduce these two lines to `self._state = bbb_gpio.read_input(self._pin)`